PWGHF/TableProducer/candidateCreatorSigmac0plusplusCascade.cxx

```diff
@@ -0,0 +1,324 @@
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file candidateCreatorSigmac0plusplusCascade.cxx
+///
+/// \note Task for Σc0,++ → Λc(→ K0sP) + π-,+ analysis
+/// \note Here the Lc from the cascade channel is obtained using the task taskLcToK0sP.cxx
+/// \author Rutuparna Rath <rrath@cern.ch>, INFN BOLOGNA and GSI Darmstadt
+/// In collaboration with Andrea Alici <aalici@cern.ch>, INFN BOLOGNA
+
+#include "CommonConstants/PhysicsConstants.h"
+#include "Common/Core/TrackSelection.h"
+#include "Common/Core/trackUtilities.h"
+#include "Common/Core/TrackSelectionDefaults.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/runDataProcessing.h"
+#include "PWGHF/Core/HfHelper.h"
+#include "PWGHF/DataModel/CandidateReconstructionTables.h"
+#include "PWGHF/DataModel/CandidateSelectionTables.h"
+
+using namespace o2;
+using namespace o2::analysis;
+using namespace o2::constants::physics;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+struct HfCandidateCreatorSigmac0plusplusCascade {
+  Configurable<double> cfgtrkMinPt{"cfgtrkMinPt", 0.15, "track min pT"};
+  Configurable<double> cfgtrkMaxEta{"cfgtrkMaxEta", 0.8, "track max Eta"};
+  Configurable<double> cfgMaxDCAxyToPVcut{"cfgMaxDCAxyToPVcut", 2.0, "Track DCAxy cut to PV Maximum"};
+  Configurable<double> cfgMaxDCAzToPVcut{"cfgMaxDCAzToPVcut", 2.0, "Track DCAz cut to PV Maximum"};
+  Configurable<double> cfgnFindableTPCClusters{"cfgnFindableTPCClusters", 120, "nFindable TPC Clusters"};
+  Configurable<double> cfgnTPCCrossedRows{"cfgnTPCCrossedRows", 70, "nCrossed TPC Rows"};
+  Configurable<double> cfgnTPCChi2{"cfgnTPChi2", 4.0, "nTPC Chi2 per Cluster"};
+  Configurable<double> cfgnITSChi2{"cfgnITShi2", 36.0, "nITS Chi2 per Cluster"};
+  Configurable<double> cfgTPCnSigmaPi{"cfgTPCnSigmaPi", 3.0, "TPC nSigma selection"};
+
+  /// Table with Σc0,++ info
+  Produces<aod::HfCandScCasBase> rowScCandidates;
+  /// Selection of candidates Λc+
+  Configurable<int> selectionFlagLc{"selectionFlagLc", 1, "Selection Flag for Lc"};
+  Configurable<double> yCandLcMax{"yCandLcMax", -1., "max. candLc. Lc rapidity"};
+  Configurable<std::vector<double>> binsPt{"binsPt", std::vector<double>{hf_cuts_sigmac_to_p_k_pi::vecBinsPt}, "pT bin limits"};
+  Configurable<LabeledArray<double>> cutsMassLcMax{"cutsMassLcMax", {hf_cuts_sigmac_to_p_k_pi::cuts[0], hf_cuts_sigmac_to_p_k_pi::nBinsPt, hf_cuts_sigmac_to_p_k_pi::nCutVars, hf_cuts_sigmac_to_p_k_pi::labelsPt, hf_cuts_sigmac_to_p_k_pi::labelsCutVar}, "Lc candidate selection per pT bin"};
+  Configurable<int> selectionFlagLcToK0sP{"selectionFlagLcToK0sP", 1, "Selection Flag for Lc"};
```

Comment:
these cuts are never used in the code, i.e. in this way you always buil SigmaC candidates with Lc candidates of whatever invariant mass. I suggest to add this cut in the code.

Reply:
In the Lc selector, we used the topo cuts to select the Lc candidates. Later I will add it.

Reply:
See my new comment, actually you do not exploit the info from the selector at the moment. In addition, with the same idea in mind, I suggest you to implement already a mass cut around Lc inv. mass to build Sc candidates, as I said already in my previous comment

Reply:
Thank you. Done.